sys/include/sema.h

```diff
@@ -33,21 +34,42 @@ extern "C" {
 /**
  * @brief   Creates semaphore statically.
  *
- * @param[in] value Initial value for the semaphore.
+ * The value for the statically created semaphore must be 0
  *
  * @return  Statically initialized semaphore.
  */
-#define SEMA_CREATE(value)      { (value), PRIORITY_QUEUE_INIT }
+#define SEMA_CREATE()           { (0), SEMA_OK, MUTEX_INIT() }
+
+/**
+ * @brief A Semaphore states.
+ */
+typedef enum {
+    SEMA_OK = 0,
+    SEMA_DESTROY,
+} sema_state_t;
 
 /**
  * @brief A Semaphore.
  */
 typedef struct {
-    volatile unsigned int value;    /**< value of the semaphore */
-    priority_queue_t queue;         /**< list of threads waiting for the semaphore */
+    unsigned int value;             /**< value of the semaphore */
+    sema_state_t state;             /**< state of the semaphore */
+    mutex_t mutex;                  /**< mutex of the semaphore */
 } sema_t;
 
 /**
+ * @brief   Get value of a semaphore.
+ *
+ * @param[in]  sema Semaphore to get the value from
+ *
+ * @return  Current value of the semaphore.
+ */
+static inline unsigned int sema_value(sema_t *sema)
```

Comment:
Actually it doesn't make a difference, so I removed it :+1:  Then there was no need to touch the `semaphore.h` file :)